### PR TITLE
chore: move to use vite as devtool

### LINKED ISSUE #928: [General] - move this project developer setup to use vite server
### What general issue would you like to create?

currently this project uses CRA developer setup (which is deprecated by meta team)
[vite](https://vitejs.dev) is more modern recommended tool for react projects

### PR DESCRIPTION
**Pull request by shootermv**
## Summary of changes

migrate project to use vite devtool because it more modern tool for react projects

## Checklist

- [x] I have read and followed the [contribution guidelines](https://github.com/freeCodeCamp/Developer_Quiz_Site/blob/main/CONTRIBUTING.md).
- [x] I have read through the [Code of Conduct](https://github.com/freeCodeCamp/Developer_Quiz_Site/blob/main/CODE_OF_CONDUCT.md) and agree to abide by the rules.
- [x] This PR is for one of the available issues and is not a PR for an issue already assigned to someone else.
- [x] My PR title has a short descriptive name so the maintainers can get an idea of what the PR is about.
- [x] I have provided a summary of my changes.

closes #928

### CODE REVIEW

**Comment by jdwilkin4** [comment]
Hey @shootermv ! 

I was able to fork your copy of the project.
When I run it locally I get an error for the `manifest.json`

Screenshot 2023-12-21 at 8 05 56 PM

So I tried my own github pages build of the project and the build failed due to issues with the `manifest.json`
Screenshot 2023-12-21 at 8 07 03 PM

Also, I was googling around and for the `gh-pages` package we are using it looks like we need to update the deploy script to point to the dist folder instead of the build folder according to this article 
https://dev.to/rashidshamloo/deploying-vite-react-app-to-github-pages-35hf

Since this is a migration, we will have to test actual builds of test sites to make sure that this PR doesn't accidentally break the production site.
I am testing everything by creating a github pages site on my forked copy of your fork.
I would suggest testing the builds on your end and seeing if you can your forked copy to build a github pages site to make sure the migration work.

Hope that helps 👍

**Comment by jdwilkin4** [comment]
Here are my settings for testing the builds to GH pages 

Screenshot 2023-12-21 at 8 15 35 PM

**Comment by shootermv** [comment]
Sorry, totally my bad!
i will look to fix it asap

**Comment by shootermv** [comment]
After looking at the tutorials you sent - i changed deployment script and was able to deploy to my gh-pages
Please look if you able to run & deploy the project now without issues...
p.s. Thank you for  guidance!

File: package.json
Comment (jdwilkin4) [inline package.json:23]: since we are using the default dist folder, this should be updated too

```suggestion
    "deploy": "gh-pages -d dist",
```

File: package.json
Comment (jdwilkin4) [inline package.json:28]: when I ran the test command, all the tests pass fine but it generates a results.json file

Screenshot 2023-12-22 at 8 18 44 AM

It would be best to add the file to the gitignore so it doesn't get committed to the repo by accident.

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
Was able to play around with it locally and left a few comments 👍

File: package.json
Comment (shootermv) [reply package.json:28]: configured tests to output to tests_results folder

File: package.json
Comment (shootermv) [reply package.json]: is there something i forgot?

Comment (jdwilkin4) [reply package.json]: No, it looks like all of the comments have been resolved.
The whole freeCodeCamp team is on Christmas break.
So I will take look at this PR after break 👍

**Review by jdwilkin4 — APPROVED** [review]
Was able to do a build and preview locally and it was working on my end.

So we should be good to go.
I vote we merge this in and see if the build goes through. 👍 

Then if there are any issues, we can always roll back a commit to the old version and debug from there and try again 👍 

I would suggest holding off on deleting this branch until we know the production build worked out

**Comment by shootermv** [comment]
> Was able to do a build and preview locally and it was working on my end.
> 
> So we should be good to go.
> I vote we merge this in and see if the build goes through. 👍 
> 
> Then if there are any issues, we can always roll back a commit to the old version and debug from there and try again 👍 
> 
> I would suggest holding off on deleting this branch until we know the production build worked out 

Totally agree

**Pull request merged**
